cosign.py

```diff
@@ -102,13 +128,18 @@ def sign_image(
         ).hexdigest()
         image_reference = f'{image_reference.ref_without_tag}@sha256{digest}'
 
-    if on_exist is OnExist.SKIP:
+    if on_exist in (OnExist.SKIP, OnExist.APPEND):
         if oci_client.head_manifest(
```

Comment:
Maybe use `exists = bool(oci_client.head_manifest(...))`. In general, I think patterns which set a variable to `True` if a condition is true and to `False` if the condition is false should be prevented if possible.

Reply:
okay, changed accordingly. this saves two lines of code, while making it very slightly less readable (imho). which is okay in such simple cases.

why do you think 
```
if <condition>:
  marker = True
else:
  marker = False
```
should be avoided? I think it is very easy to read, albeit not very elegant (-> KISS)

Reply:
I think this is just duplicated code, since the condition is equivalent to the value of `marker` and I think duplicated code should be avoided unless it has any extra value. If you think it is more readable to choose your variant, I'm fine to keep it since the extra value is a better readability then.